nimble/host/services/audio/pacs/src/ble_audio_svc_pacs.c

```diff
@@ -0,0 +1,428 @@
+#include "host/ble_gatt.h"
+#include "host/ble_audio_codec.h"
+#include "services/pacs/ble_audio_svc_pacs.h"
+
+static uint32_t ble_svc_audio_pacs_sink_audio_locations;
+static uint32_t ble_svc_audio_pacs_source_audio_locations;
+static struct available_ctx {
+    uint16_t conn_handle;
+    uint16_t ble_svc_audio_pacs_avail_sink_contexts;
+    uint16_t ble_svc_audio_pacs_avail_source_contexts;
+}  ble_svc_audio_pacs_avail_contexts[MYNEWT_VAL(BLE_MAX_CONNECTIONS)] = {
+    {
+        .conn_handle = BLE_HS_CONN_HANDLE_NONE,
+    }
+};
+static uint16_t ble_svc_audio_pacs_sup_sink_contexts;
+static uint16_t ble_svc_audio_pacs_sup_source_contexts;
+
+static struct ble_gap_event_listener ble_pacs_listener;
+
+struct pac_read_cb_arg {
+    uint8_t flags;
+    struct os_mbuf *om;
+    uint8_t pac_count;
+};
+static int
+ble_svc_audio_pacs_access(uint16_t conn_handle, uint16_t attr_handle,
+                          struct ble_gatt_access_ctxt *ctxt, void *arg);
+
+static const struct ble_gatt_svc_def ble_svc_audio_pacs_defs[] = {
+    { /*** Service: Published Audio Capabilities Service (PACS) */
+        .type = BLE_GATT_SVC_TYPE_PRIMARY,
+        .uuid = BLE_UUID16_DECLARE(BLE_SVC_AUDIO_PACS_UUID16),
+        .characteristics = (struct ble_gatt_chr_def[]) { {
+            .uuid = BLE_UUID16_DECLARE(BLE_SVC_AUDIO_PACS_CHR_UUID16_SINK_PAC),
+            .access_cb = ble_svc_audio_pacs_access,
+            .flags = BLE_GATT_CHR_F_READ_ENC,
+        } , {
+            .uuid = BLE_UUID16_DECLARE(
+                BLE_SVC_AUDIO_PACS_CHR_UUID16_SINK_AUDIO_LOCATIONS),
+            .access_cb = ble_svc_audio_pacs_access,
+            .flags = BLE_GATT_CHR_F_READ_ENC,
+        } , {
+            .uuid = BLE_UUID16_DECLARE(
+                BLE_SVC_AUDIO_PACS_CHR_UUID16_SOURCE_PAC),
+            .access_cb = ble_svc_audio_pacs_access,
+            .flags = BLE_GATT_CHR_F_READ_ENC
+        } , {
+            .uuid = BLE_UUID16_DECLARE(
+                BLE_SVC_AUDIO_PACS_CHR_UUID16_SOURCE_AUDIO_LOCATIONS),
+            .access_cb = ble_svc_audio_pacs_access,
+            .flags = BLE_GATT_CHR_F_READ_ENC,
+        }, {
+            .uuid = BLE_UUID16_DECLARE(
+                BLE_SVC_AUDIO_PACS_CHR_UUID16_AVAILABLE_AUDIO_CONTEXTS),
+            .access_cb = ble_svc_audio_pacs_access,
+            .flags = BLE_GATT_CHR_F_READ_ENC | BLE_GATT_CHR_F_NOTIFY,
+        }, {
+            .uuid = BLE_UUID16_DECLARE(
+                BLE_SVC_AUDIO_PACS_CHR_UUID16_SUPPORTED_AUDIO_CONTEXTS),
+            .access_cb = ble_svc_audio_pacs_access,
+            .flags = BLE_GATT_CHR_F_READ_ENC,
+        }, {
+            0, /* No more characteristics in this service */
+        }
+        }
+    },
+    {
+        0, /* No more services. */
+    },
+};
+
+static int
+codec_record_to_pacs_entry(struct ble_audio_codec_record *record, void *arg)
+{
+    struct pac_read_cb_arg *cb_arg = (struct pac_read_cb_arg *)arg;
+    int rc;
+
+    if (!(record->flags & cb_arg->flags)) {
```

Comment:
As you might wan to filter the so called type or direction of the CAP, please define and use mask
`BLE_AUDIO_CODEC_FLAG_SOURCE | BLE_AUDIO_CODEC_FLAG_SINK`